internal/util/get/get.go

```diff
@@ -67,41 +68,142 @@ func (c Command) Run() error {
 	// define where we are going to clone the package from
 	r := &git.RepoSpec{OrgRepo: c.Repo, Path: c.Directory, Ref: c.Ref}
 
-	defaultRef, err := gitutil.DefaultRef(c.Repo)
+	err := cloneAndCopy(r, c.Destination, c.Name, c.Clean)
 	if err != nil {
 		return err
 	}
 
-	// clone the repo to a tmp directory.
-	// delete the tmp directory later.
-	err = ClonerUsingGitExec(r, defaultRef)
+	if err = c.fetchRemoteSubpackages(); err != nil {
+		return errors.Wrap(err)
+	}
+	return nil
+}
+
+// fetchRemoteSubpackages goes through the root package and its subpackages
+// and fetches any remote subpackages referenced. It will also handle situations
+// where a remote subpackage references other remote subpackages.
+func (c Command) fetchRemoteSubpackages() error {
+	// Create a stack to keep track of all Kptfiles that needs to be checked
+	// for remote subpackages.
+	stack := newStack()
+
+	paths, err := pathutil.DirsWithFile(c.Destination, kptfile.KptFileName, true)
+	if err != nil {
+		return err
+	}
+	for _, p := range paths {
+		stack.push(p)
+	}
+
+	for stack.len() > 0 {
+		p := stack.pop()
+		kf, err := kptfileutil.ReadFile(p)
+		if err != nil {
+			return err
+		}
+
+		remoteSubPkgDirs := make(map[string]bool)
+		for i := range kf.Subpackages {
+			sp := kf.Subpackages[i]
+
+			if _, found := remoteSubPkgDirs[sp.LocalDir]; found {
+				return fmt.Errorf("multiple remote subpackages with localDir %q", sp.LocalDir)
+			}
+			remoteSubPkgDirs[sp.LocalDir] = true
+
+			gitInfo := sp.Git
+			localPath := filepath.Join(p, sp.LocalDir)
+
+			_, err = os.Stat(localPath)
+			if err != nil && !os.IsNotExist(err) {
+				return err
```

Comment:
So this error means the remote subpackage which is being fetched is already present in the directory as a local subpackage? Can this error be more clear including the steps to correct it so that user know what is the actual error and the fix.